apps/discord-bot/src/triggers/cta-post.ts

```diff
@@ -0,0 +1,280 @@
+            console.log(`Adding ${role?.first()?.name} member, ${user.displayName}`);
+          }
+        }
+      });
+    });
+    return roleReactions;
+  }
+
+  private async activeCTAThread(msg: Message): Promise<PublicThreadChannel | undefined> {
+    if (msg.hasThread && msg.thread?.type === ChannelType.PublicThread && !msg.thread?.archived) {
+      return msg.thread;
+    }
+  }
+
+  private async createCTAThread(msg: Message): Promise<PublicThreadChannel | undefined> {
+
+    if (msg.channel.type === ChannelType.GuildText) {
+      const thread = msg.startThread({
+        // set as the title of the message
+
+        name: 'CTA Completion by Region',
+        autoArchiveDuration: ThreadAutoArchiveDuration.OneWeek,
+        reason: 'Tracking CTA participation.',
+      });
+
+      const threadChannel = await thread;
+      threadChannel.send({
+        content: `Mark your completion of the CTA by reacting with ${finishedEmoji}.\n\n A chart will be posted in an hour, and updated every hour until the thread is archived, showing the number of completions per region.`,
+        flags: [MessageFlags.SuppressNotifications]
+      });
+      msg.react(finishedEmoji)
+
+      return thread
+    }
+
+    return undefined
+  }
+
+  // start a reaction collector on the given message
+  private async startCTAReactionCollector(msg: Message, thread: PublicThreadChannel): Promise<void> {
+    const filter = (reaction, user) => reaction.emoji.name === finishedEmoji && user.id != msg.client.user.id
+    const collector = msg.createReactionCollector({ filter, time: oneHour });
+    const roleReactions = await this.getFinishedReactions(msg)
+
+    collector.on('collect', async (reaction, user) => {
+      const member = msg.guild?.members.fetch(user.id)
+      if (member != undefined) {
+
+        console.log(`Got ${reaction.emoji.name} from ${(await member).displayName}`)
+
+        const memberRoles = (await member).roles.cache;
+        const role = memberRoles.filter(r => regionRoles.includes(r.name));
+        const memberRegionRole = role.first()?.name;
+
+        if (memberRegionRole != undefined) {
+          if (!roleReactions[memberRegionRole]) {
+            roleReactions[memberRegionRole] = [];
+          }
+          if (!roleReactions[memberRegionRole]?.includes(user.id)) {
+            roleReactions[memberRegionRole]?.push(user.id);
+            console.log(`Adding ${role.first()?.name} member, ${user.displayName}`);
+          }
+        }
+
+        this.sendChartToThread(msg, thread, roleReactions);
+      }
+    });
+
+    collector.on('end', collected => {
+      console.log(`Collected ${collected.size} reactions.`);
+      this.sendChartToThread(msg, thread, roleReactions);
+    });
+  }
+
+  private async deleteLastChart(tc: PublicThreadChannel): Promise<void> {
+    const threadMessages = await tc.messages.fetch({ limit: 1 });
+    const latest = threadMessages.first();
```

Comment:
These two lines seem like they're assuming the chart will be the only message in the thread, but I don't see anything enforcing that in the creation of the thread. Will it be enforced through permissions?